sequencer/src/state.rs

```diff
@@ -94,14 +98,17 @@ pub fn validate_proposal(
     let receipts = fetch_fee_receipts(parent.l1_finalized, proposal.l1_finalized);
     for FeeInfo { account, amount } in receipts {
         // Get the balance in order to add amount, ignoring the proof.
-        match fee_merkle_tree.universal_lookup(account) {
-            LookupResult::Ok(balance, _) => fee_merkle_tree
-                .update(account, balance.add(amount))
-                .unwrap(),
-            // Handle `NotFound` and `NotInMemory` by initializing
-            // state.
-            _ => fee_merkle_tree.update(account, amount).unwrap(),
-        };
+        let lookup = fee_merkle_tree.universal_lookup(account);
+        if matches!(lookup, LookupResult::NotInMemory) {
+            let view = todo!(); // where to get the view number?
+            block_on(peers.remember_account_balance(view, account, fee_merkle_tree));
+        }
+        fee_merkle_tree
+            .update_with(account, |balance| {
+                Some(balance.cloned().unwrap_or_default().add(amount))
+            })
+            .expect("update_with succeeds");
+        // TODO: can this really not fail?
```

Comment:
I don't think so, b/c both `Some` or `None` cases are handled in the closure. I feel that this can be written more  succinctly, though, by *remembering* the account balance in the `Ok(NotInMemory)` case of the match on `update_with` (as you've done ~line 143).